### ISSUE TITLE
how to make libusb installed?

### ISSUE THREAD

**Issue by guanguanboy** [libuvc/libuvc#99]
Getting and Building libuvc
Prerequisites: You will need libusb and CMake installed.
To build, you can just run these shell commands:
git clone https://github.com/ktossell/libuvc
cd libuvc
mkdir build
cd build
cmake ..
make && sudo make install

I just wonder how to make  libusb installed?

**Comment by mcuee** [comment]
libusb has binary download and it contains binary for MSVC and MinGW 32/64bit. You can download the binary from github release page or libusb Sourceforge.
https://github.com/libusb/libusb/releases
https://sourceforge.net/projects/libusb/files/libusb-1.0/

If you use infrastructure like MSYS2, then it has the libusb binary inside and it has git/cmake also.

**Comment by ArifSohaib** [comment]
There seems to be a mistake in the libuvc_internal.h file.
The include says #include <libusb.h>
But according to https://askubuntu.com/questions/629619/how-to-install-libusb
after installing libusb-dev, it needs to be included like #include <libusb-1.0/libusb.h>
It now builds fine for me.
Note that I am kind of new to C++ and build systems so I may be wrong. (I know C++ but I mostly use Visual Studio or Qt)

**Comment by guanguanboy** [comment]
@mcuee  thx a lot。

**Comment by mcuee** [comment]
@ArifSohaib If you use CMake or pkg-config to detect libusb, then you do not need to use <<libusb-1.0/libusb.h>, in fact, that is the recommended way.

**Comment by ArifSohaib** [comment]
@mcuee Thanks. I would have preferred not to change any existing code to avoid problems down the road but as I said, I am kind of new to make/build systems. Is this https://cmake.org/cmake/help/v3.0/command/find_library.html the method to find the library? And is it supposed to go in the CMakeLists or Makefile?

**Comment by mcuee** [comment]
Yes that is supposed to be the way to use CMake to find a library.

I think the fork here offer the option to build under Windows using Visual C++, it has the option to specify the libusb directory. 
https://github.com/pupil-labs/libuvc
https://github.com/ktossell/libuvc/blob/master/CMakeLists.txt

But then I think that fork is not that good to support non-VC build infrastructure like MSYS2.
I am not an expert in CMake. I know libftdi has the CMake files to find libusb, which may be more cross-platform.
http://developer.intra2net.com/git/?p=libftdi;a=blob;f=cmake/FindUSB1.cmake
